plugins/woocommerce-blocks/assets/js/editor-components/switch-to-classic-shortcode-button/index.tsx

```diff
@@ -0,0 +1,165 @@
+		block === 'woocommerce/cart'
+			? __( 'Switched to classic cart.', 'woocommerce' )
+			: __( 'Switched to classic checkout.', 'woocommerce' );
+
+	const { getBlocks } = useSelect( ( select ) => {
+		return {
+			getBlocks: select( blockEditorStore ).getBlocks,
+		};
+	}, [] );
+
+	const selectClassicShortcodeBlock = () => {
+		const classicShortcodeBlock = findBlock( {
+			blocks: getBlocks(),
+			findCondition: ( foundBlock: BlockInstance ) =>
+				foundBlock.name === 'woocommerce/classic-shortcode',
+		} );
+
+		if ( classicShortcodeBlock ) {
+			selectBlock( classicShortcodeBlock.clientId );
+		}
+	};
+
+	return (
+		<>
+			<Button
+				variant={ 'secondary' }
+				onClick={ () => {
+					recordEvent( 'switch_to_classic_shortcode_click', {
+						shortcode:
+							block === 'woocommerce/checkout'
+								? 'checkout'
+								: 'cart',
+					} );
+					openModal();
+				} }
+			>
+				{ switchButtonLabel }
+			</Button>
+			{ isOpen && (
+				<Modal
+					size="medium"
+					title={ switchButtonLabel }
+					onRequestClose={ closeModal }
+					className="wc-blocks-switch-to-classic-shortcode-modal-content"
+				>
+					<ModalContent blockType={ block } />
+					<TabbableContainer className="wc-blocks-switch-to-classic-shortcode-modal-actions">
+						<Button
+							variant="primary"
+							isDestructive={ true }
+							onClick={ () => {
+								replaceBlock(
+									clientId,
+									createBlock(
+										'woocommerce/classic-shortcode',
+										{
+											shortcode:
+												block === 'woocommerce/checkout'
+													? 'checkout'
+													: 'cart',
+										}
+									)
+								);
+								recordEvent(
+									'switch_to_classic_shortcode_confirm',
+									{
+										shortcode:
+											block === 'woocommerce/checkout'
+												? 'checkout'
+												: 'cart',
+									}
+								);
+								selectClassicShortcodeBlock();
+								createInfoNotice( snackbarLabel, {
+									actions: [
+										{
+											label: __( 'Undo', 'woocommerce' ),
+											onClick: () => {
+												undo();
+												recordEvent(
+													'switch_to_classic_shortcode_undo',
+													{
+														shortcode:
+															block ===
+															'woocommerce/checkout'
+																? 'checkout'
+																: 'cart',
+													}
+												);
+											},
+										},
+									],
+									type: 'snackbar',
+								} );
+								closeModal();
+							} }
+						>
+							{ __( 'Switch', 'woocommerce' ) }
+						</Button>{ ' ' }
+						<Button
+							variant="secondary"
+							onClick={ () => {
+								recordEvent(
+									'switch_to_classic_shortcode_cancel',
+									{
+										shortcode:
+											block === 'woocommerce/checkout'
+												? 'checkout'
+												: 'cart',
+									}
+								);
+								closeModal();
+							} }
+						>
+							{ __( 'Cancel', 'woocommerce' ) }
+						</Button>
+					</TabbableContainer>
+				</Modal>
```

Comment:
This block is pretty long, which affects the readability. What are your thoughts on splitting it into smaller parts? 

> [!IMPORTANT]
> The code snippets below were generated using ChatGPT-4 to showcase potential enhancements in the readability of the <ModalContent> section. Please note that they have not been thoroughly reviewed and are not intended for direct implementation without further validation.

The main function at the end of the file could look like this:

```ts
<ModalContent blockType={block} />
    <TabbableContainer className="wc-blocks-switch-to-classic-shortcode-modal-actions">
        <SwitchButton />
        {' '}
        <CancelButton />
    </TabbableContainer>
</Modal>
```

Here, we're simply calling the components `<SwitchButton>` and `<CancelButton />`. These components could look like this:

```ts
const SwitchButton = () => (
    <Button variant="primary" isDestructive={true} onClick={handleSwitchClick}>
        { __( 'Switch', 'woocommerce' ) }
    </Button>
);

const CancelButton = () => (
    <Button variant="secondary" onClick={handleCancelClick}>
        { __( 'Cancel', 'woocommerce' ) }
    </Button>
);
```

Finally, we'd define functions for the `onClick` events. These functions could look like this:

```ts
const handleSwitchClick = () => {
    replaceBlock(
        clientId,
        createBlock('woocommerce/classic-shortcode', {
            shortcode: block === 'woocommerce/checkout' ? 'checkout' : 'cart',
        })
    );
    recordEvent('switch_to_classic_shortcode_confirm', {
        shortcode: block === 'woocommerce/checkout' ? 'checkout' : 'cart',
    });
    selectClassicShortcodeBlock();
    createInfoNotice(snackbarLabel, {
        actions: [
            {
                label: __( 'Undo', 'woocommerce' ),
                onClick: handleUndoClick,
            },
        ],
        type: 'snackbar',
    });
    closeModal();
};

const handleUndoClick = () => {
    undo();
    recordEvent('switch_to_classic_shortcode_undo', {
        shortcode: block === 'woocommerce/checkout' ? 'checkout' : 'cart',
    });
};

const handleCancelClick = () => {
    recordEvent('switch_to_classic_shortcode_cancel', {
        shortcode: block === 'woocommerce/checkout' ? 'checkout' : 'cart',
    });
    closeModal();
};
```